CudaQ Kernel Test Failure

pytest yields the following failure.
```
    def test_cudaq_kernel_requires_annotation_matrices():
        cudaq = pytest.importorskip("cudaq")
    
        @cudaq.kernel
        def bell_pair():
            q = cudaq.qvector(2)
            h(q[0])  # noqa: F821  # pyright: ignore[reportUndefinedVariable]
            cx(q[0], q[1])  # noqa: F821  # pyright: ignore[reportUndefinedVariable]
    
>       with pytest.raises(
            ValueError,
            match="At least one of m2dets or m2obs must be provided for CUDA-Q kernels",
        ):
E       Failed: DID NOT RAISE <class 'ValueError'>
```

In https://github.com/QuEraComputing/bloqade-lanes/blob/main/python/bloqade/lanes/device.py , we see

```
if is_cudaq_kernel(logical_kernel):
    logical_squin_kernel: ir.Method = cudaq_to_squin(logical_kernel)

    # Default to Steane [[7,1,3]] annotation matrices when not provided
    if m2dets is None and m2obs is None:
        num_qubits = len(_find_qubit_ssas(logical_squin_kernel))
        m2dets = steane7_m2dets(num_qubits)
        m2obs = steane7_m2obs(num_qubits)
```

So likely, this test is deprecated; we should not expect an exception.

--- a/python/tests/test_device.py
+++ b/python/tests/test_device.py
@@ -245,22 +245,6 @@ def test_append_annotations_to_kernel_with_terminal_measure(
         assert all(isinstance(b, bool) for obs in result.observables for b in obs)
 
 
-def test_cudaq_kernel_requires_annotation_matrices():
-    cudaq = pytest.importorskip("cudaq")
-
-    @cudaq.kernel
-    def bell_pair():
-        q = cudaq.qvector(2)
-        h(q[0])  # noqa: F821  # pyright: ignore[reportUndefinedVariable]
-        cx(q[0], q[1])  # noqa: F821  # pyright: ignore[reportUndefinedVariable]
-
-    with pytest.raises(
-        ValueError,
-        match="At least one of m2dets or m2obs must be provided for CUDA-Q kernels",
-    ):
-        GeminiLogicalSimulator().task(bell_pair)
-
-
 @pytest.mark.parametrize(
     "use_dets, use_obs",
     [(True, True), (True, False), (False, True)],

